### PR TITLE
change: [M3-8468] - Improve action menu styling

### PR DESCRIPTION
**Pull request by harsh-akamai**
## Description 📝
When a user opens the LinodeActionMenu by using their mouse, the first option remain highlighted, even after the user is hovering over a separate option. This can be confusing for our users and negatively impacts accessibility.

It has also been reported that the LinodeActionMenu could use additional visual enhancements during the active state. In particular, when visiting the Linodes landing page, the top of the menu can be difficult to identify for some users because it's aligned with the "Last Backup" column of the Linodes table. This is particularly more evident when a user is using dark mode settings.

## Changes  🔄
- Added an eventHandler to change the `<MenuItem/>` state to focus status `onMouseEnter`
- Changed the active state color for `<MenuItem` while in dark mode

## Target release date 🗓️
N/A

## Preview 📷

- On hovering over menu items

| Before  | After   |
| ------- | ------- |
|image| image
|

- Visual enhacement

| Before highlight color change  | After highlight color change   |
| ------- | ------- |
| image| image|

### Prerequisites
- Open the website in dark theme  

### Verification steps
- Navigate to `localhost:3000/linodes` and click on the `<KebabIcon/>` for any one of the Linodes
- Ensure that the first item('Power Off` in this case) doesn't remain highlighted when you are hovering over other options
- Verify that the `Last Backup` Column of the Linodes table is able to be differentiated from the top of the option menu

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/.changeset/pr-10964-changed-1726673831823.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/manager": Changed
+---
+
+Improve ActionMenu styling ([#10964](https://github.com/linode/manager/pull/10964))
```
Comment (dwiley-akamai) [inline packages/manager/.changeset/pr-10964-changed-1726673831823.md:5]:
```suggestion
Improve ActionMenu styling and accessibility ([#10964](https://github.com/linode/manager/pull/10964))
```

**Review by dwiley-akamai — APPROVED** [review]
Verified focus/hover state and color differentiation improvements ✅

**Review by bnussman-akamai — COMMENTED** [review]
I'm wondering if this change could be made in the theme rather than in the component itself. 

https://github.com/linode/manager/blob/518a6634373ab35b359435dc0ff741d1a23857c3/packages/manager/src/foundations/themes/light.ts#L1009

**Review by cpathipa — APPROVED** [review]
Confirming the action menu styles. No regressions were found in dark or light mode.

**Comment by bnussman-akamai** [comment]
I'm not sure @harsh-akamai 

I'm mostly just concerned with one-off styling of the `MenuItem`. I think in an ideal world, this issue would be fixed in the theme or design-language-system rather than by a stray `sx`. 

If the `design-language-system` color doesn't make sense and needs updating, we can make the change there!

**Comment by harsh-akamai** [comment]
Changed the color for active state as the one I chose in the initial commit coincided with the disabled state color.

| Initial change  |  New change   |
| ------- | ------- |
|image| image|

**Pull request merged**